### DISCUSSION TITLE
Encrypted password to infinity datasource

### DISCUSSION THREAD

**Discussion by MarielaPages** [grafana/grafana-infinity-datasource#527] [Q&A]
I would like to know if the infinity datasource accepts encrypted passwords in its configuration, and is so, if there is a command or mechanism to encrypt those passwords that the infinity datasource would accept as it.

**Comment by yesoreyeram** [comment]
I don't understand the question. Can you provide more details on what you are trying to achieve?

**Reply by MarielaPages** [reply]
Sorry, yes. 

I'm trying to configure the data source infinity by passing encrypted passwords instead of the real ones. The thing is that clients may not want to pass their passwords as they are so that we can configue the datasource for them, they want to pass encrypted ones. So they were asking if there is a way in which they can encrypt them, pass them to us like that and we can use them to configure the data source. I didn't see this option in the authentication methods, but I was wondering if there was a way to do so. 

Thanks for the response!

**Reply by yesoreyeram** [reply]
Doesn't make sense to me. If your password is encrypted how the plugin will decrypt? If your server can decrypt, you can use encrypted password as it is in the plugin. The plugin have nothing to do with the passwords other than passing it to the server.
